zoo/src/main/scala/com/intel/analytics/zoo/common/NNContext.scala

```
@@ -40,7 +45,81 @@ object NNContext {
     val bigdlConf = Engine.createSparkConf(conf)
     val sc = SparkContext.getOrCreate(bigdlConf)
     Engine.init
+    checkSparkVersion(sc)
+    checkScalaVersion()
     sc
   }
+
+  private def checkVersion(
+                            runtimeVersion: String,
+                            compileTimeVersion: String,
+                            project: String): Unit = {
+    val Array(runtimeMajor, runtimeFeature, runtimeMaintance) =
+      runtimeVersion.split("\\.").map(_.toInt)
+    val Array(compileMajor, compileFeature, compileMaintance) =
+      compileTimeVersion.split("\\.").map(_.toInt)
+    if (!(runtimeMajor == compileMajor && runtimeFeature == compileFeature)) {
```

add an option to ignore the error and just report a warning.